src/gc/gc.d

```diff
@@ -2122,6 +2308,12 @@ struct Gcx
                 auto pool = findPool(p);
                 if (pool)
                 {
+                    static if (precise) if (p1pool)
+                    {
+                        size_t p1off = cast(size_t)(p1 - cast(void**)p1pool.baseAddr);
```

Comment:
I just tried it: without further optimizations, moving the code above `p = *p1` slows down marking by up to 10%.